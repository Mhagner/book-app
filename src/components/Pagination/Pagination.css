.pagination__container {
    display: flex;
    justify-content: center;
    padding: 1.6rem;
    margin-top: 1.6rem;
    align-items: center;
}

.pagination__button {
    position: relative;
    padding: 1.6rem;
    border-radius: 50%;
    width: 3.2rem;
    height: 3.2rem;
    border: 1px solid rgba(51, 51, 51, 0.2);
    background: transparent;
    transition: all 0.2s;
}

.pagination__button:hover {
   background-color: var(--color-primary-light);
   transform: scale(1.08);
}

.pagination__button img {
    position: absolute;
    top: 1.2rem;
}

.pagination__arrow-left {
    left: 1.2rem;
}

.pagination__arrow-right {
    right: 1.2rem;
}

.pagination__count {
    padding: 0 1.8rem;
    font-size: 1.2rem;
}


.pagination__count strong {
    margin: 0 0.5rem;
}

@media (min-width: 1026px) {
    .pagination__container {
        justify-content: flex-end;
        padding: 0;
    }
}