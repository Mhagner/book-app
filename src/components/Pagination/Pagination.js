import './Pagination.css';
import ArrowLeft from '../../assets/images/arrow-left.svg'
import ArrowRight from '../../assets/images/arrow-right.svg'

const Pagination = ({ 
    handlePreviousPage, 
    handleNextPage, 
    totalPages, 
    currentPage 
}) => {
    return (
        <div data-testid="pagination" className="pagination__container">
            <button
                className="pagination__button"
                onClick={handlePreviousPage}
            >
                <img
                    className="pagination__arrow-left"
                    src={ArrowLeft}
                    alt="arrow left"
                />
            </button>
            <p className="pagination__count">
                PÃ¡gina
                <strong>{currentPage}</strong> de
                <strong>{totalPages ? Math.round(totalPages) : 0}</strong>
            </p>
            <button
                className="pagination__button"
                onClick={handleNextPage}
            >
                <img
                    className="pagination__arrow-right"
                    src={ArrowRight}
                    alt="arrow right"
                />
            </button>
        </div>
    )
}

export default Pagination;