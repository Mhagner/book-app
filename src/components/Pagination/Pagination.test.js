import { render } from '@testing-library/react'

import Pagination from './Pagination'

describe('Pagination component', () => {
    const renderComponent = (props) => {
        return render(<Pagination {...props}/>)
    }

    it('Componente should defined', () => {
        const { getByTestId } = renderComponent()

        const element = getByTestId('pagination')

        expect(element).toBeDefined()
    })

    it('Componente should render', () => {

        const { getByTestId } = renderComponent()

        const element = getByTestId('pagination')

        expect(element).toMatchSnapshot()
    })

    it('Componente should render pagination current 5 and total page 50', () => {
        const props = {
            totalPages: 50,
            currentPage: 5
        }

        const { getByTestId } = renderComponent(props)

        const element = getByTestId('pagination')

        expect(element).toMatchSnapshot()
    })

})