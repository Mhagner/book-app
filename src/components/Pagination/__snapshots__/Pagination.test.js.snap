// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Componente should render 1`] = `
<div
  class="pagination__container"
  data-testid="pagination"
>
  <button
    class="pagination__button"
  >
    <img
      alt="arrow left"
      class="pagination__arrow-left"
      src="arrow-left.svg"
    />
  </button>
  <p
    class="pagination__count"
  >
    Página
    <strong />
     de
    <strong>
      0
    </strong>
  </p>
  <button
    class="pagination__button"
  >
    <img
      alt="arrow right"
      class="pagination__arrow-right"
      src="arrow-right.svg"
    />
  </button>
</div>
`;

exports[`Pagination component Componente should render pagination current 5 and total page 50 1`] = `
<div
  class="pagination__container"
  data-testid="pagination"
>
  <button
    class="pagination__button"
  >
    <img
      alt="arrow left"
      class="pagination__arrow-left"
      src="arrow-left.svg"
    />
  </button>
  <p
    class="pagination__count"
  >
    Página
    <strong>
      5
    </strong>
     de
    <strong>
      50
    </strong>
  </p>
  <button
    class="pagination__button"
  >
    <img
      alt="arrow right"
      class="pagination__arrow-right"
      src="arrow-right.svg"
    />
  </button>
</div>
`;
